src/Components/RecommendationsPage/RecommendationsTable.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { Button, Skeleton } from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, ThProps, Tr } from '@patternfly/react-table';
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+
+import { RECOMMENDATIONS_DETAIL_ROOT } from '../../Helpers/constants';
+import { Recommendation } from './types';
+
+const RecommendationsTable = ({ recommendations, perPage, showSkeleton }) => {
+  const columnNames = {
+    description: 'Name',
+    publish_date: 'Modified',
+    systems: 'Systems',
+    remediation: 'Remediation',
+    run: 'Run',
+  };
+
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(null);
+
+  const getSortableRowValues = (rec: Recommendation): (string | number | Date)[] => {
+    const { description, publish_date, systems, remediation } = rec;
+    return [description, publish_date, systems, remediation];
+  };
+
+  if (activeSortIndex !== null) {
+    recommendations.sort((a, b) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (typeof aValue === 'number') {
+        // Numeric sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as number) - (bValue as number);
+        }
+        return (bValue as number) - (aValue as number);
+      } else {
+        // String sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as string).localeCompare(bValue as string);
+        }
+        return (bValue as string).localeCompare(aValue as string);
+      }
+    });
+  }
+  const getSortParams = (columnIndex: number): ThProps['sort'] => {
+    return {
+      sortBy: {
+        index: activeSortIndex as number,
+        direction: activeSortDirection as 'asc' | 'desc',
+        defaultDirection: 'asc',
+      },
+      onSort: (_event, index, direction) => {
+        setActiveSortIndex(index);
+        setActiveSortDirection(direction);
+      },
+      columnIndex,
+    };
+  };
+
+  const loadinSkeletons = [...Array(perPage)].map((_, index) => (
+    <Tbody key={index}>
+      <Tr key={index}>
+        {[...Array(Object.keys(columnNames).length + 1)].map((_, index) => (
+          <Td key={index}>
+            <Skeleton />
+          </Td>
+        ))}
+      </Tr>
+    </Tbody>
+  ));
```

Comment:
Also this we can move to the helpers function (please then change it in the tasks table :smile: )